jobserver/middleware.py

```diff
@@ -0,0 +1,52 @@
+from django.shortcuts import redirect
+
+
+class RequireNameMiddleware:
+    """
+    Redirect users without `fullname` set to the settings page
+
+    We get a User's username from GitHub when they register with the site.  If
+    their GitHub profile has a name set we also get that, however we have many
+    (~50% at time of writing) users who have not set that.  We want to require
+    all users have their name set.  This middleware redirects any user with an
+    empty `fullname` value to their settings page.
+
+    We exempt various URL prefixes which don't make sense to do this on, eg if
+    we're not expecting traditional users there or it doesn't make sense for
+    the UX (in the case of applications).
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        exemptions = (
+            "/__debug__",
+            "/__reload__",
+            "/api",
+            "/applications",
+            "/apply",
+            "/complete",
+            "/disconnect",
+            "/favicon.ico",
+            "/login",
+            "/logout",
+            "/robots.txt",
+            "/settings",
+            "/staff",
+            "/status",
```

Comment:
Hmm, this might bite us if we change urls for any of these. But hopefully this is a short term thing!

Minor suggestion - order this most-popular to least, which will make the `startswith()` clause below more efficient overall. OTTOMH I think `/api` and `/staff` and `/robots.txt` are likely to be very common.